File: chap2/Oui.java
// Ces programmes sont sous licence CeCILL-B V1.
class Oui {

  public static void main (String [] args) {
    String s;
    int longueur, n;
    s = "un oui ou un non ce n'est pourtant pas la même chose";
    longueur = Isn.stringLength(s);
    n = 0;
    while (n <= longueur-3 && !(Isn.stringEqual(Isn.stringNth(s,n),"o")
                                && Isn.stringEqual(Isn.stringNth(s,n+1),"u")
                                && Isn.stringEqual(Isn.stringNth(s,n+2),"i"))) {
      n = n + 1;}
    if (n > longueur - 3) {
      System.out.println("pas de oui");} 
    else {
      System.out.println(n);}}}

File: chap2/Oui.c
// Ces programmes sont sous licence CeCILL-B V1.
#include <stdio.h>
#include <string.h>

int main()
{
   int longueur, n;
   char s[] = "un oui ou un non ce n'est pourtant pas la même chose";
   longueur = strlen(s);
   n = 0;
   while (n <= longueur-3 && s[n] != 'o' && s[n+1] != 'u' && s[n+2] != 'i')
   {
      n = n + 1;
   }
   if (n > longueur - 3) 
   {
      printf("pas de oui\n");
   } 
   else 
   {
      printf("%d\n",n);
   }
}
